sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -471,17 +474,73 @@ public <T> CosmosPagedFlux<T> queryItems(String query, Class<T> classType) {
      */
     public Mono<Void> openConnectionsAndInitCaches() {
 
-        if(isInitialized.compareAndSet(false, true)) {
-            return withContext(context -> openConnectionsAndInitCachesInternal()
+        if (isInitialized.compareAndSet(false, true)) {
+
+            CosmosContainerIdentity cosmosContainerIdentity = new CosmosContainerIdentity(this.database.getId(), this.id);
+            ProactiveContainerInitConfig proactiveContainerInitConfig = new ProactiveContainerInitConfigBuilder(Arrays.asList(cosmosContainerIdentity))
+                    .setProactiveConnectionRegions(1)
+                    .build();
+
+            return withContext(context -> openConnectionsAndInitCachesInternal(proactiveContainerInitConfig)
                                             .flatMap(openResult -> {
                                                 logger.info("OpenConnectionsAndInitCaches: {}", openResult);
                                                 return Mono.empty();
                                             }));
         } else {
-            logger.warn(
-                    String.format(
-                        "OpenConnectionsAndInitCaches is already called once on Container %s, no operation will take place in this call",
-                        this.getId()));
+            logger.warn("OpenConnectionsAndInitCaches is already called once on Container {}, no operation will take place in this call", this.getId());
+            return Mono.empty();
+        }
+    }
+
+    /**
+     *  Best effort to initialize the container by warming up the caches and connections to a specified no. of regions from the
+     *  preferred list of regions.
+     *
+     *  Depending on how many partitions the container has, the total time needed will also change. But generally you can use the following formula
+     *  to get an estimated time:
+     *  If it took 200ms to establish a connection, and you have 100 partitions in your container
+     *  then it will take around (100 * 4 / CPUCores) * 200ms to open all connections after get the address list
```

Comment:
Does this formula include number of remote regions as well or connection to these are/will be opened in parallel?
I.e. if we have 3 regions (numProactiveConnectionRegions = 3) with 1000 partitions would it mean 
(3*1000*4/CPU)*200ms? For CPU = 4 this would mean 600s (10 minutes)? I think CPU cores to thread factor and thread priority are already settings (non-public) but wondering if there is another way to somehow speed up the connection creation? Cases of 1000s of partitions with 3+ regions already exist.

Reply:
Corrected the formula - your sample above would actually take about

(3 * 1000 * 4/ (10 * CPUs)) * 200 ms - so for 4 CPU roughly 60 seconds - given that 200ms avg. connection opening time is rather high I think this estimate is pretty pessimistic. We can also evaluate whether 1 core can open more than 10 connections concurrently - the concurrency chosen is probably also pretty restrictive - but this will definitely need some additional testing.